packages/product/src/loaders/connection.ts

```diff
@@ -62,7 +83,29 @@ async function loadDefault({ database, container }) {
   const entities = Object.values(ProductModels) as unknown as EntitySchema[]
   const orm = await DALUtils.mikroOrmCreateConnection(database, entities)
 
+  return orm.em.fork()
+}
+
+async function loadShared({ container, logger }) {
+  const sharedConnection = container.resolve(
+    ContainerRegistrationKeys.PG_KNEX_CONNECTION_REGISTRATION_KEY,
+    {
+      allowUnregistered: true,
+    }
+  )
+  if (!sharedConnection) {
+    ;(logger ?? console)?.warn(
+      "The Product module is setup to use a shared resources but no shared connection is present. A new connection will be created"
+    )
+  }
```

Comment:
suggestion: If a shared connection is expected at this point, might make sense to throw an error here. It indicates that something went wrong and needs to be handled.